src/Microsoft.AspNetCore.SignalR/SignalRAppBuilderExtensions.cs

```diff
@@ -32,9 +32,14 @@ public HubRouteBuilder(SocketRouteBuilder routes)
             _routes = routes;
         }
 
-        public void MapHub<THub>(string path) where THub : Hub
+        public void MapHub<THub>(string path) where THub : Hub<IClientProxy>
         {
             _routes.MapEndpoint<HubEndPoint<THub>>(path);
         }
+
+        public void MapHub<THub, TClient>(string path) where THub : Hub<TClient>
```

Comment:
I started with `DefaultPresenceManager<THub>` was `DefaultPresenceManager<THub, TClient>` which required this but this indeed did not work (I think I hit: https://github.com/aspnet/SignalR/issues/48).  Fixing https://github.com/aspnet/SignalR/issues/48 did not seem straightforward and was not the main point of this PR. I will remove this overload for now.